core/app/models/spree/wallet.rb

```diff
@@ -61,6 +61,11 @@ def default_wallet_payment_source=(wallet_payment_source)
       raise Unauthorized, "wallet_payment_source #{wallet_payment_source.id} does not belong to wallet of user #{user.id}"
     end
 
+    # Do not update the payment source if the passed source is already default
+    if default_wallet_payment_source.try(:id) == wallet_payment_source.try(:id)
```

Comment:
It's probably better to also not use `.try(:id)`. `default_wallet_payment_source == wallet_payment_source`